### ISSUE TITLE
UserTaskRun Search with status/userGroup/userId not working

### ISSUE THREAD

**Issue by coltmcnealy-lh** [littlehorse-enterprises/littlehorse#805]
```
{
  "id": {
    "wfRunId": {
      "id": "1d627294f6a64ac58ecb58462af29c0f"
    },
    "userTaskGuid": "4a6ffaa5400c4e27a15336b0cd07f614"
  },
  "userTaskDefId": {
    "name": "it-request",
    "version": 0
  },
  "userGroup": "testGroup",
  "userId": "jumbo",
  "results": {
    "justification": {
      "str": "algo"
    },
    "requestedItem": {
      "str": "algo "
    }
  },
  "status": "DONE",
  "events": [
    {
      "time": "2024-05-13T19:44:58.759Z",
      "assigned": {
        "oldUserGroup": "testGroup",
        "newUserId": "jumbo",
        "newUserGroup": "testGroup"
      }
    }
  ],
  "scheduledTime": "2024-05-13T19:43:30.930Z",
  "nodeRunId": {
    "wfRunId": {
      "id": "1d627294f6a64ac58ecb58462af29c0f"
    },
    "threadRunNumber": 0,
    "position": 1
  },
  "epoch": 3
}
```

When I search by status - DONE, I have it on the results.
When I search by userId - jumbo, I have it on the results.
When I search by userGroup - testGroup. I have it on the results.
When I search by userId + userGroup. I have it on the results.
When I search by userId + userGroup + status It does not come in the results :thinking_face:

```
lhctl search userTaskRun --userId jumbo --userGroup testGroup --userTaskStatus DONE
```

**Comment by coltmcnealy-lh** [comment]
In addition to the problem reported above by @diablouma , it seems that the following query also does not work:

```
lhctl search userTaskRun --userId obiwan  --userTaskStatus DONE --userGroup testGroup --userTaskDefName it-request
```

It was also reported by Nelson that the problem is not `lhctl` but rather in the GRPC API.

**Comment by coltmcnealy-lh** [comment]
There are four Attributes + the "created_at" in the UserTaskRun:
* `userId`
* `userGroup`
* `userTaskDefName`
* `status`

The way the `rpc SearchUserTaskRun` is structured, every one of those Attributes is optional. So you can specify any of them.  That means we have `2^4` == `16` potential combinations of Attributes for any given `rpc SearchUserTaskRun`, which means that we need  16 Tags for each `UserTaskRun` (depending on whether userId and userGroup are present). Currently, `UserTaskRunModel#getIndexConfigurations()` only specifies 10 indices, which means 6 of the cases are not covered. Nelson discovered one of those cases today.

We need to add the missing 6 indices unfortunately. Implementing a partial-prefix-scan would be quite error-prone and might actually not cover all of the cases.

**Comment by eduwercamacaro** [comment]
Yes, you are right. we need to add those tags.

**Comment by coltmcnealy-lh** [comment]
Fixed by #806
